docs/workflow.md

```diff
@@ -0,0 +1,49 @@
+# Workflow
+
+In here we describe the general workflow guidelines the NuGet developer/contributor.
+
+## Basics
+
+### Pull requests & Code reviews
+
+To help ensure that only the highest quality code makes its way into the project, all code changes need to be submitted to GitHub as PRs.
+
+In general a PR should be approved by the Subject Matter Expert (SME) of that code. For example, a change to the Banana project should be signed off by `@Monkey`, and not by `@Giraffe`. If you don't know the SME, someone on the team will help you identify them. Of course, sometimes it's the SME who is making a change, in which case a secondary person will have to sign off on the change (e.g. `@JuniorMonkey`).
+
+To commit the PR to the repo use the GitHub `Squash and Merge` button. We can't stress this enough. Always use `Squash and Merge` unless an exception is explicitly stated in this document.
+
+- *Do* favor having more than 1 reviewer.
+- *Do not* merge too quickly. Wait for at least 24h after the last significant changes before merging unless the change is urgent.
+- *Do* address all feedback. Not necessarily by accepting it, but by reaching a resolution with the reviewer. All comments need to be marked as resolved before merging.
+- *Do* use GitHub's tooling. Re-request review after all feedback has been addressed.
+
+#### Draft Pull Requests
+
+Draft pull requests are allowed, but should have a clear plan for transition to a review ready pull request.
+Draft pull requests will be closed within 30 days.
+
+### Branching strategy
+
+The active development branch in our repo is `dev`. What we ship comes from the `release-majorminorx` branches.
+
+### Solution and project folder structure and naming
+
+The NuGet.Client repo currently has only one solution file named `NuGet.sln`. We do not want/need to have more than one solution file. 
+If deemed necessary by the team, we can consider solution filters at a future point.
+
```

Comment:
Suggestions to add:

- Don't add references to external, non-NuGet packages, unless you have a strong justification for doing so.
- If you need a dependency to a NuGet library, prefer a `ProjectReference` over a `PackageReference` to that NuGet package
- Honor the target frameworks in `common.project.props`

Reply:
> Don't add references to external, non-NuGet packages, unless you have a strong justification for doing so.

That's a good idea, I think it deserves its own section mentioning source-build etc.

Reply:
All these are good ideas, but maybe not appropriate for this section. I will add them to a list of future improvements.